### PR TITLE
Closeby Workflows with TICL Variations in runTheMatrix

### PR DESCRIPTION
**Pull request by rovere**
#### PR description:

This PR does the following:

   * Fix the Closeby snippet to match the name with the actual particles generated
   * Add Closeby Workflow to the upgrade session of runTheMatrix
   * Add 2 variations to this workflow that will add TICL to the full Reco (.52) or run TICL reconstruction alone (.51)

This PR, in order to be fully tested, needs #27382.

Since runTheMatrix is a little outside of my area of expertise, please provide alternative solutions if you are not satisfied with this PR.

Finally, given the deadline of the 6th Patatrack hackathon next week, it would be nice to have an IB with this and #27382 integrated.

#### PR validation:

```
runTheMatrix.py -l limited -i all --ibeos
```

and

```
runTheMatrix.py -w upgrade -l 29093.51,29093.52
```

### CODE REVIEW

File: Configuration/PyReleaseValidation/python/relval_upgrade.py [hunk 1]
```diff
@@ -89,6 +89,16 @@ def makeStepName(key,frag,step,suffix):
                 elif '2018' in key:
                     workflows[numWF+upgradeSteps['pixelTrackingOnly']['offset']] = [ upgradeDatasetFromFragment[frag], stepList['pixelTrackingOnly']]
 
+            # special workflows for HGCAL/TICL
+            if (upgradeDatasetFromFragment[frag]=="CloseByParticleGun"):
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/relval_upgrade.py]: can you restrict this to 2023 workflows? (We don't need to run TICL for Phase 1)

**Comment by kpedro88** [comment]
@rovere if you want these workflows to be tested in IBs, you should add the WF numbers to relval_2023.py

File: Configuration/PyReleaseValidation/python/relval_upgrade.py (hunk 1 above)
Comment (rovere) [reply Configuration/PyReleaseValidation/python/relval_upgrade.py]: Sure!

**Comment by rovere** [comment]
> @rovere if you want these workflows to be tested in IBs, you should add the WF numbers to relval_2023.py

Thanks, I just add 1 TICL workflow to the D41 section

**Comment by kpedro88** [comment]
please test workflow 29093.52

**Comment by rovere** [comment]
> please test workflow 29093.52

Without #27382 it will fail in creating the python configuration and will also fail in the assert.
That's why I explicitly required #27382.
I tested locally with #27382 on top and it works. Not sure how to proceed in this case.

**Comment by kpedro88** [comment]
please abort

**Comment by kpedro88** [comment]
please test workflow 29093.52 with #27382

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
@prebello, @zhenhu, @efeyazgan, @pgunnell, @agrohsje, @alberto-sanchez, @qliphy could you please check this PR?

**Comment by prebello** [comment]
+1
@rovere are these new workflows aimed to be tested only in IB?

**Comment by qliphy** [comment]
+1

**Comment by rovere** [comment]
> +1
> @rovere are these new workflows aimed to be tested only in IB?

For the time being yes. The plan is to request some specific sample in the form of RelVal at some point in the (near) future.

**Comment by kpedro88** [comment]
@fabiocos can we merge this now?

**Comment by fabiocos** [comment]
+1

**Comment by fwyzard** [comment]
@kpedro88 @fabiocos please do not use the modifiers .51, .52, .53 .

They are being used in the Patatrack branch (after we had to move from the .7 etc.) we were using before.

**Comment by rovere** [comment]
Indeed, I overlooked the numbering. Will shift it in a coming PR (unless @kpedro88 or someone else wants to take care of that)

**Comment by kpedro88** [comment]
@fwyzard maybe it would be useful for you to submit a PR updating upgradeWorkflowComponents.py with your usage of these numbers, even if they aren't applied to any actual workflows, so that we can keep track of reserved values more easily.

**Comment by fabiocos** [comment]
@rovere @kpedro88 so what would be your alternative proposal for this PR?

**Comment by kpedro88** [comment]
We just need to change the numbering to something like .101 and .102. This is not urgent, just a point for the longer term to avoid the conflict with patatrack.

**Pull request merged**